File: workback/src/utils/httpClient.js
class HttpClient {
  /**
   * GET请求
   */
  async get(url, options = {}) {
    try {
      const response = await fetch(url, {
        method: 'GET',
        ...options
      });

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }

      return await response.json();
    } catch (error) {
      throw new Error(`GET请求失败: ${error.message}`);
    }
  }

  /**
   * POST请求
   */
  async post(url, data = null, options = {}) {
    try {
      let body = null;
      let defaultHeaders = {};

      if (data) {
        // 如果data是字符串（表单数据），直接使用
        if (typeof data === 'string') {
          body = data;
          defaultHeaders['Content-Type'] = 'application/x-www-form-urlencoded; charset=UTF-8';
        } else {
          // 否则JSON序列化
          body = JSON.stringify(data);
          defaultHeaders['Content-Type'] = 'application/json';
        }
      }

      const response = await fetch(url, {
        method: 'POST',
        body: body,
        headers: {
          ...defaultHeaders,
          ...options.headers
        },
        ...options
      });

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }

      return await response.json();
    } catch (error) {
      throw new Error(`POST请求失败: ${error.message}`);
    }
  }

  /**
   * PUT请求
   */
  async put(url, data, options = {}) {
    try {
      const response = await fetch(url, {
        method: 'PUT',
        body: JSON.stringify(data),
        headers: {
          'Content-Type': 'application/json',
          ...options.headers
        },
        ...options
      });

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }

      return await response.json();
    } catch (error) {
      throw new Error(`PUT请求失败: ${error.message}`);
    }
  }

  /**
   * DELETE请求
   */
  async delete(url, options = {}) {
    try {
      const response = await fetch(url, {
        method: 'DELETE',
        ...options
      });

      if (!response.ok) {
        throw new Error(`HTTP error! status: ${response.status}`);
      }

      return await response.json();
    } catch (error) {
      throw new Error(`DELETE请求失败: ${error.message}`);
    }
  }
}

module.exports = new HttpClient(); 

File: workbackpy/utils/http_client.py
import requests
from requests.adapters import HTTPAdapter
from urllib3.util.retry import Retry
import json

class HttpClient:
    """HTTP 客户端工具类"""
    
    def __init__(self, timeout=30, retries=3):
        """
        初始化 HTTP 客户端
        
        Args:
            timeout: 请求超时时间（秒）
            retries: 重试次数
        """
        self.timeout = timeout
        self.session = requests.Session()
        
        # 配置重试策略
        retry_strategy = Retry(
            total=retries,
            status_forcelist=[429, 500, 502, 503, 504],
            method_whitelist=["HEAD", "GET", "OPTIONS"]
        )
        adapter = HTTPAdapter(max_retries=retry_strategy)
        self.session.mount("http://", adapter)
        self.session.mount("https://", adapter)
    
    def get(self, url, params=None, headers=None):
        """GET 请求"""
        try:
            response = self.session.get(
                url, 
                params=params, 
                headers=headers, 
                timeout=self.timeout
            )
            response.raise_for_status()
            return response.json() if response.content else None
        except requests.exceptions.RequestException as e:
            raise Exception(f"GET 请求失败: {str(e)}")
    
    def post(self, url, data=None, json_data=None, headers=None):
        """POST 请求"""
        try:
            response = self.session.post(
                url, 
                data=data, 
                json=json_data, 
                headers=headers, 
                timeout=self.timeout
            )
            response.raise_for_status()
            return response.json() if response.content else None
        except requests.exceptions.RequestException as e:
            raise Exception(f"POST 请求失败: {str(e)}")
    
    def put(self, url, data=None, json_data=None, headers=None):
        """PUT 请求"""
        try:
            response = self.session.put(
                url, 
                data=data, 
                json=json_data, 
                headers=headers, 
                timeout=self.timeout
            )
            response.raise_for_status()
            return response.json() if response.content else None
        except requests.exceptions.RequestException as e:
            raise Exception(f"PUT 请求失败: {str(e)}")
    
    def delete(self, url, headers=None):
        """DELETE 请求"""
        try:
            response = self.session.delete(
                url, 
                headers=headers, 
                timeout=self.timeout
            )
            response.raise_for_status()
            return response.json() if response.content else None
        except requests.exceptions.RequestException as e:
            raise Exception(f"DELETE 请求失败: {str(e)}")
    
    def close(self):
        """关闭会话"""
        self.session.close() 